ompi/mca/coll/base/coll_base_alltoallv.c

```diff
@@ -66,8 +66,15 @@ mca_coll_base_alltoallv_intra_basic_inplace(const void *rbuf, const int *rcounts
     size = ompi_comm_size(comm);
     rank = ompi_comm_rank(comm);
 
-    ompi_datatype_type_size(rdtype, &max_size);
-    max_size *= rcounts[rank];
+    for (i = 0, max_size = 0 ; i < size ; ++i) {
+        if (i == rank) {
+            continue;
+        }
+        ompi_datatype_type_size(rdtype, &packed_size);
```

Comment:
I know the datatype cache its size, but we can avoid the unnecessary call to ompi_datatype_type_size by calling it once outside the loop

Reply:
fixed, but `ompi_datatype_type_size` is supposed to be inlined

Reply:
a linearly with rank number of unnecessary calls.